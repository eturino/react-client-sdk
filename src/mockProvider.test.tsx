import React, { FunctionComponent } from 'react';
import { create } from 'react-test-renderer';
import MockProvider from './mockProvider';
import { useFlags } from './index';

const MockApp: FunctionComponent = () => {
  const flags = useFlags();

  return <div>Value: {`${flags.testFlag}`}</div>;
};

describe('MockProvider', () => {
  it('allows the usage of `useFlags` with the given mocked flags', () => {
    const LaunchDarklyApp = (
      <MockProvider flags={{ testFlag: 'my-test' }}>
        <MockApp />
      </MockProvider>
    );
    const component = create(LaunchDarklyApp);
    expect(component).toMatchSnapshot();
  });
});
