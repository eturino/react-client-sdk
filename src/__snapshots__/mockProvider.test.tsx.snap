// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MockProvider allows the usage of \`useFlags\` with the given mocked flags 1`] = `
<div>
  Value: 
  my-test
</div>
`;
